src/pfcp/3gpp_29.244.hpp

```diff
@@ -803,11 +803,14 @@ class pfcp_fteid_ie : public pfcp_ie {
 class pfcp_network_instance_ie : public pfcp_ie {
  public:
   std::string network_instance;
+  std::string network_instance_dotted;
 
   //--------
   explicit pfcp_network_instance_ie(const pfcp::network_instance_t& b)
       : pfcp_ie(PFCP_IE_NETWORK_INSTANCE) {
     network_instance = b.network_instance;
+    pfcp_ie::string_to_dotted(network_instance, network_instance_dotted);
+    network_instance = network_instance_dotted;
     tlv.set_length(network_instance.size());
```

Comment:
Finally network_instance = network_instance_dotted and both are declared
std::string network_instance;
std::string network_instance_dotted;
One string may be enough ?

Reply:
By doing like this then your string will be always dotted, for printing it, you will have to do a reverse conversion, could be fine.
The alternative solution is to call string_to_dotted only when serializing and dotted_to_string when de-serializing, this has the drawback to be take more CPU if called very often.